harlogger/sniffers.py

```diff
@@ -0,0 +1,218 @@
+
+    def should_keep(self, entry_hash: EntryHash) -> bool:
+        if self._filters.get('pids') and entry_hash.pid in self._filters.get('pids'):
+            return False
+
+        if self._filters.get('images') and entry_hash.image in self._filters.get('images'):
+            return False
+
+        if self._filters.get('process_names') and entry_hash.process_name in self._filters.get(
+                'process_names'):
+            return False
+        return True
+
+
+class SnifferProfile(SnifferStrategy):
+    """
+    Sniff using CFNetowrkDiagnostics.mobileconfig profile.
+
+    This requires the specific Apple profile to be installed for the sniff to work.
+    """
+
+    def __init__(self, lockdown: LockdownClient, filters: Mapping = None, response: bool = True, request: bool = True,
+                 color: bool = True):
+        self._response = response
+        self._color = color
+        self._request = request
+        self._shown_list = []
+        self._lockdown = lockdown
+        self._os_trace_service = OsTraceService(self._lockdown)
+        self._filters = filters if filters else {}
+
+    def sniff(self):
+        for entry in self._os_trace_service.syslog():
+            if entry.label is None or entry.label.subsystem != 'com.apple.CFNetwork' or \
+                    entry.label.category != 'Diagnostics':
+                continue
+
+            entry_hash = EntryHash(entry.pid,
+                                   posixpath.basename(entry.filename),
+                                   os.path.basename(entry.image_name),
+                                   '')
+
+            if not self.should_keep(entry_hash):
+                continue
+
+            self.show(entry_hash, entry)
+
+    def show(self, entry_hash: EntryHash, entry: Any) -> None:
+
+        lines = entry.message.split('\n')
+        if len(lines) < 2:
+            return
+        buf = ''
+
+        if lines[1].strip().startswith('Protocol Enqueue: request') and self._request:
+            # request
+            print('➡️   ', end='')
+            fields = self._parse_fields(entry.message)
+            buf += f'{fields["Message"]}\n'
+            for name, value in fields.items():
+                if name in ('Protocol Enqueue', 'Request', 'Message'):
+                    continue
+                buf += f'{name}: {value}\n'
+
+        elif lines[1].strip().startswith('Protocol Received: request') and self._response:
+            # response
+            print('⬅️   ', end='')
+            fields = self._parse_fields(entry.message)
+            buf += f'{fields["Response"]} ({fields["Protocol Received"]})\n'
+            for name, value in fields.items():
+                if name in ('Protocol Received', 'Response'):
+                    continue
+                buf += f'{name}: {value}\n'
+
+        if buf:
+            if self._color:
+                print(highlight(buf, HttpLexer(), TerminalTrueColorFormatter(style='autumn')))
```

Comment:
make `automn` into const, or maybe even give the user an option to pass a different value using `click.Choice()` and make `automn` the default